### PR TITLE
may well be the culprit of #1461

### PR DESCRIPTION
**Pull request by GillesDuvert**

### CODE REVIEW

**Comment by GillesDuvert** [comment]
please test this (clever?) workaround.

**Comment by alaingdl** [comment]
Same situation see from here (u2204 & Debian 10)

BUT I may have some clues for you : `window & window, 2` gave 2 windows of different sizes
Same if : `plot, findgen(10) & window,2` or `window, 0 & plot, findgen(10) & window,2`

If I call again `window, 0` the new generated window does have the same size than window 2
(and is higher that first window 0)

Another problem : `print, !d.x_size, !d.y_size` gives same size but windows don't have same size !

```
GDL> window, 0
GDL> print, !d.x_size, !d.y_size
         960         540
GDL> window, 1
GDL> print, !d.x_size, !d.y_size
         960         540
```

HTH

**Comment by alaingdl** [comment]
Pfu, long time I don't used that : xwininfo

First window is :   Width: 950  Height: 506
Second window is :  Width: 960 Height: 540

What I did : 
```
GDL> plot, findgen(10)
GDL> window, 1,xs=100, ys=200
GDL> window, 2

and in another window :
$ xwininfo -tree -root | grep GDL
 
        0x300003b "GDL 2": ("gdl" "Gdl")  960x540+5+29  +6+56
        0x300002a "GDL 1": ("gdl" "Gdl")  100x200+5+29  +1815+875
        0x3000013 "GDL 0": ("gdl" "Gdl")  950x506+5+29  +955+56


```

and the same with `gdl --no-use-wx`
```
        0x5000011 "GDL 2": ()  960x540+5+29  +6+56
        0x500000a "GDL 1": ()  100x200+5+29  +1815+875
        0x5000001 "GDL 0": ()  960x540+5+29  +955+56
```

**Comment by alaingdl** [comment]
I spend some time making `cout` in gdlwxstream.cpp, devicewx.hpp plotting_windows.cpp

Comments !
-- the sizes (X,Y) of the windows are OK (values as expected) in every WOpen() I checked
-- I put a sleep before WOpen, clearly the window is "plotted" 2 times
-- When copying the first window with w0=TVRD() and doing a TV, w0 in a wider plot, the whole plot is OK (xaxis are here)

 question : 
-- why when calling just `plot, findgen(10)` without having a window before the code don't go through the 
window() as defined in  plotting_windows.cpp ? (then I began to say the issue comes from plplot ?!)

some #define gdlBORDER_ are defined in gdlwidgets.hpp. Any chance they interact with the first plot (then being overwritten by our values ?)

**Comment by GillesDuvert** [comment]
Hm all this seems related with windows that need to produce a resize event when they are resized (since you need to have your plot resized if you resize manually the window, there is even a special case where the window is larger than the screen size see #708). Unfortunately creating the window in the first place produce sizes events (even a lot of size events on gnome (?)) which 1) may 'clear' the window and 2) give the size of the window+decoration, but not the size of the plot window itself, and decoration sizes are not well known by wxWidgets.

Basically this behaviour has not changed since a few years so I expect this size discrepancy to be there in gdl 0.1.x . Besides, it works the same on OSX and Windows, and that never caused any problem (?)

I guess I'll have to install an Debian 10 VM to check this, as I do not have all those pesky problems. 

BTW, you refer to window size problems, but is the first bug (first plot not seeen) still present?

**Comment by GillesDuvert** [comment]
OK, installed a Debian 10 in a VM.
With exactly this PR's code, indeed the first plot is not seen until one clicks back (gives focus back to) the terminal.

**Comment by GillesDuvert** [comment]
Normally this should solve the 'first plot not seen' problem.
Then the size of windows, if persist, can be solved easily I think.

**Comment by GillesDuvert** [comment]
Actually, not quite.
There are some side-effects.
Basically the wxwidgets eventloop is not always present to handle the resizes, or refresh.

**Comment by GillesDuvert** [comment]
The test fails on ubuntu (no display) but the PR is OK on ALL platforms, solving #1461 but also #1463 .
... And installers are OK too.

**Comment by alaingdl** [comment]
I am not sure to understand what I have to check !
OK, compiled on U22.04.
The first plot is now OK
(but the size of the window for the first plot is wrong)

Now testing on OSX

**Comment by alaingdl** [comment]
On OSX, I got :

```
 In file included from /Users/alaingdl/GDL/gdl-1.0.1-git230105CMake/src/graphicsdevice.cpp:34:
In file included from /Users/alaingdl/GDL/gdl-1.0.1-git230105CMake/src/deviceps.hpp:22:
/Users/alaingdl/GDL/gdl-1.0.1-git230105CMake/src/plotting.hpp:258:54: error: unknown type name 'PROJTYPE'
  DDoubleGDL* GDLgrGetProjectPolygon(GDLGStream * a, PROJTYPE ref, DStructGDL* map, DDoubleGDL *lons, DDoubleGDL *lats, DDoubleGDL *zVal, bool isRadians, bool const doFill, bool const dolines, DLongGDL *&conn);
                                                     ^

**Comment by alaingdl** [comment]
Incredible !! On OSX, this PR, with the script + install : launching is OK, then plotting is OK _and the size of the windows is OK !_ Marvellous !! (but why my U2204 is broken ?!)

**Comment by GillesDuvert** [comment]
Works on Debian 10 (on a VM here), my Mageia (of course), Windows 10 and OSX...
